catalog/transformer/catalog-transformer-pptx/src/test/java/ddf/catalog/transformer/input/pptx/PptxInputTransformerTest.java

```diff
@@ -0,0 +1,241 @@
+
+public class PptxInputTransformerTest {
+
+    private static List<String> files = Arrays.asList("/45541_Footer.pptx",
+            "/45541_Header.pptx",
+            "/45545_Comment.pptx",
+            "/49386-null_dates.pptx",
+            "/51187.pptx",
+            "/54542_cropped_bitmap.pptx",
+            "/56812.pptx",
+            "/KEY02.pptx",
+            "/OverlappingRelations.pptx",
+            "/SampleShow.pptx",
+            "/WithMaster.pptx",
+            "/ae.ac.uaeu.faculty_nafaachbili_GeomLec1.pptx",
+            "/alterman_security.pptx",
+            "/at.ecodesign.www_downloads_Vertiefungsvortrag_elektronik.pptx",
+            "/backgrounds.pptx",
+            "/layouts.pptx",
+            "/pie-chart.pptx",
+            "/pptx2svg.pptx",
+            "/prProps.pptx",
+            "/present1.pptx",
+            "/sample.pptx",
+            "/sample_pptx_grouping_issues.pptx",
+            "/shapes.pptx",
+            "/table_test.pptx",
+            "/testPPT.pptx",
+            "/themes.pptx",
+            "/with_japanese.pptx",
+            "/hello-world-signed.pptx",
+            "/hello-world-unsigned.pptx",
+            "/ms-office-2010-signed.pptx");
+
+    /*
+    This was used to just create a thumbnail for manual testing.
+
+    @Test
+    public void test1() throws IOException, CatalogTransformerException {
+        PptxInputTransformer t = new PptxInputTransformer();
+        try (InputStream is = PptxInputTransformerTest.class.getResourceAsStream("/KEY02.pptx")) {
+            Metacard mc = t.transform(is);
+            try (FileOutputStream fos = new FileOutputStream("thumbnail.jpg")) {
+                fos.write(mc.getThumbnail());
+            }
+        }
+    }
+    */
+
+    private final InputTransformer inputTransformer = new TikaInputTransformer(null);
+
+    private InputStream res(String res) {
+        return PptxInputTransformerTest.class.getResourceAsStream(res);
+    }
+
+    @Test(expected = CatalogTransformerException.class)
+    public void testBadCopy() throws IOException, CatalogTransformerException {
+        try {
+            PptxInputTransformer t = new PptxInputTransformer(inputTransformer);
+            InputStream is = Mockito.mock(InputStream.class);
+            Mockito.when(is.read(Mockito.any()))
+                    .thenThrow(new IOException());
+            t.transform(is);
+        } catch (CatalogTransformerException e) {
+            Assert.assertTrue(e.getMessage()
+                    .contains("Could not copy bytes of content message."));
+            throw e;
+        }
+    }
+
+    @Test(expected = CatalogTransformerException.class)
+    public void testTransformNullInput() throws IOException, CatalogTransformerException {
+        PptxInputTransformer t = new PptxInputTransformer(inputTransformer);
+        t.transform(null);
+    }
+
+    @Test
+    public void testBunchOfPptxFiles() throws IOException, CatalogTransformerException {
```

Comment:
I could probably remove this test now. I had grabbed a bunch of test PPTX files from Apache POI and was looping through them to make sure they all generated a thumbnail and didn't throw any exceptions. At the time, I didn't know what kind of edge cases in PPTX might pop up.